src/groupeddataframe/groupeddataframe.jl

```diff
@@ -572,6 +603,14 @@ function Base.haskey(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return haskey(gd, Tuple(key))
 end
 
+function Base.haskey(gd::GroupedDataFrame, key::AbstractDict{T, S}) where {T<:Union{Symbol, <:AbstractString}, S}
```

Comment:
it seems to me that this should be enough
```suggestion
function Base.haskey(gd::GroupedDataFrame, key::AbstractDict{<:Union{Symbol, AbstractString}})
```